master/internal/api_experiment.go

```diff
@@ -3040,3 +3040,111 @@ func (a *apiServer) DeleteTensorboardFiles(
 
 	return &apiv1.DeleteTensorboardFilesResponse{}, nil
 }
+
+func getPachydermConfig(config map[string]interface{}, experimentID int32) (map[string]interface{}, error) {
+	integrationConfig, ok := config["integration"].(map[string]interface{})
+	// this should not return if proper defaults are used.
+	if !ok {
+		return nil, api.NotFoundErrs("'integration' config for experiment", fmt.Sprint(experimentID), true)
+	}
+
+	pachydermConfig, ok := integrationConfig["pachyderm"].(map[string]interface{})
+	if !ok {
+		return nil, api.NotFoundErrs("'pachyderm' integration config for experiment", fmt.Sprint(experimentID), true)
+	}
+	return pachydermConfig, nil
+}
+
+func (a *apiServer) GetPachydermRepoURL(
```

Comment:
meta: why would the user call this over just getting the config themselves and doing this logic? i think you could at least swap the `a.GetExperiment` call out for something that just retrieves the config.

long run, if we're have apiServer calling other apiServer endpoints, it's going to get confusing. we may need to separate out some experiments stuff into it's own internal service.

Reply:
No I agree. I had originally suggested to product that if no validation was being done (and even if validation is being done, it should be done during config validation rather than every time this endpoint is called), then the desired URL could be created on the frontend by following the same expected pattern in this endpoint + pulling the experiment config. 

Realistically, by updating expconf to formalize the integration values there would be no real need for the addition of this endpoint.

I'm going to bring this up again with product & the frontend folks.